### PR TITLE
PRC-101: add ravelin proxy

### PR DESCRIPTION
**Pull request by philhack**
## What?
Adds an alternative client for calling the Ravelin API. It is done by proxying the API call through the [Cleudo](https://github.com/deliveroo/cluedo) service.

Sample of newing up the client
```
c = Ravelin::ProxyClient.new(base_url: 'http://cluedo.deliveroo.dev', username:'foo', password:'bar')
```

Usage is still the same
```
c.send_event(name: :order, payload: {customer_id: 'cus_123', order: { order_id: 'ord_123' }})
```

## Why?
[Cleudo](https://github.com/deliveroo/cluedo) will proxy API calls to Ravelin and also log the request/responses out in a unified logging format to franzalytics -> snowflake.
Cleudo will transform into an API that will make calls to Ravelin and other 3rd party fraud providers.

### CODE REVIEW

File: spec/ravelin/client_spec.rb [hunk 1]
```diff
@@ -12,28 +12,6 @@
 
   let(:client) { described_class.new(api_key: 'abc') }
 
-  shared_context 'event setup and stubbing' do
```
Comment (joshua-levin-deliveroo) [inline spec/ravelin/client_spec.rb:15]: I might be lacking context, should we be deleting these tests?

File: lib/ravelin/client.rb [hunk 2]
```diff
@@ -118,5 +119,31 @@ def faraday_options
...
+        headers: {
+          'Content-Type'  => 'application/json; charset=utf-8'.freeze,
+          'User-Agent'    => "Ravelin Proxy RubyGem/#{Ravelin::VERSION}".freeze
+        }
+      }
+    end
+
+  end
+
+  class ProxyClient < Client
```
Comment (joshua-levin-deliveroo) [inline lib/ravelin/client.rb:135]: I'm wondering if we should move this to its own file?

File: spec/ravelin/client_spec.rb (hunk 1 above)
Comment (joshua-levin-deliveroo) [inline spec/ravelin/client_spec.rb:15]: I might be missing something, it's not clear to me why these tests should be deleted?

File: spec/ravelin/proxy_client_spec.rb [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+    let(:tag) do
+      double('tag', name: 'ping', serializable_hash: { "customerId" => '123', "tagNames" => ['foo', 'bar'] })
+    end
+
+    before do
+      allow(Ravelin::Tag).to receive(:new).and_return(tag)
+    end
+
+    it 'calls Ravelin with correct headers and body' do
+      stub = stub_request(:delete, 'http://127.0.0.1:8020/ravelinproxy/v2/tag/customer?customerId=123&tagName=foo,bar').
```
Comment (joshua-levin-deliveroo) [inline spec/ravelin/proxy_client_spec.rb:242]: Nit: I see the url `http://127.0.0.1:8020` repeated a lot, there may be an opportunity to tidy it with a variable.  This is just a passing observation, I'm not passionate about it.

Comment (philhack) [reply spec/ravelin/proxy_client_spec.rb:242]: agreed. I'll see what I can do.

There is no place like `127.0.0.1`

Comment (joshua-levin-deliveroo) [reply spec/ravelin/proxy_client_spec.rb:242]: https://media.giphy.com/media/tnYri4n2Frnig/giphy.gif

Comment (joshua-levin-deliveroo) [reply spec/ravelin/proxy_client_spec.rb:242]: image

Comment (joshua-levin-deliveroo) [reply spec/ravelin/proxy_client_spec.rb:242]: image

File: lib/ravelin/client.rb (hunk 2 above)
Comment (philhack) [reply lib/ravelin/client.rb]: fixed in: 07482e17f94355d4943572a957e7bf9e3380f46d

File: spec/ravelin/proxy_client_spec.rb (hunk 3 above)
Comment (philhack) [reply spec/ravelin/proxy_client_spec.rb]: fixed in 8823f8377a809681cb0588b34088e146c72bc49e

**Pull request merged**